scripts/lint_prowjobs.go

```diff
@@ -0,0 +1,279 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	prowapi "k8s.io/test-infra/prow/config"
+	yaml "sigs.k8s.io/yaml"
+)
+
+type PresubmitConstants struct {
+	Bucket                   string
+	Cluster                  string
+	ServiceAccountName       string
+	DefaultMakeTarget        string
+	CniMakeTarget            string
+	HelmMakeTarget           string
+	ReleaseToolingMakeTarget string
+}
+
+type PostsubmitConstants struct {
+	Bucket                       string
+	Cluster                      string
+	DefaultServiceAccountName    string
+	AthensHelmServiceAccountName string
+	DocsServiceAccountName       string
+	MakeTarget                   string
+}
+
+type Presubmits struct {
+	PresubmitJobs map[string][]prowapi.Presubmit `json:"presubmits,omitempty"`
+}
+
+type Postsubmits struct {
+	PostsubmitJobs map[string][]prowapi.Postsubmit `json:"postsubmits,omitempty"`
+}
+
+var presubmitErrors map[string][]string
+var postsubmitErrors map[string][]string
+var presubmitErrorCount, postsubmitErrorCount = 0, 0
+
+func main() {
+	var presubmitType Presubmits
+	var postsubmitType Postsubmits
+	presubmitFiles := []string{}
+	postsubmitFiles := []string{}
+	presubmitConstants := PresubmitConstants{
+		Bucket:                   "s3://prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp",
+		Cluster:                  "prow-presubmits-cluster",
+		ServiceAccountName:       "presubmits-build-account",
+		DefaultMakeTarget:        "build",
+		CniMakeTarget:            "release",
+		HelmMakeTarget:           "verify",
+		ReleaseToolingMakeTarget: "test/build",
+	}
+	postsubmitConstants := PostsubmitConstants{
+		Bucket:                       "s3://prowdataclusterstack-316434458-prowbucket7c73355c-1n9f9v93wpjcm",
+		Cluster:                      "prow-postsubmits-cluster",
+		DefaultServiceAccountName:    "postsubmits-build-account",
+		AthensHelmServiceAccountName: "charts-build-account",
+		DocsServiceAccountName:       "docs-build-account",
+		MakeTarget:                   "release",
+	}
+
+	pullBaseSha := os.Getenv("PULL_BASE_SHA")
+	pullPullSha := os.Getenv("PULL_PULL_SHA")
+	cmdOut, err := exec.Command(`git`, "diff", "--name-only", pullBaseSha, pullPullSha).Output()
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "There was an error running the git command: ", err)
+		os.Exit(1)
+	}
+	output := string(cmdOut)
+	diffFiles := strings.Fields(output)
+	for _, file := range diffFiles {
+		if strings.Contains(file, "presubmits") {
+			presubmitFiles = append(presubmitFiles, file)
+		}
+		if strings.Contains(file, "postsubmits") {
+			postsubmitFiles = append(postsubmitFiles, file)
+		}
+	}
```

Comment:
I'd suggest pulling all the git diff +output parsing out into a separate function. Inputs being git root, base sha, pull sha, output being two lists of files and an error